Sorting/RadixSort.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+    Radix sort
+    ----------
+    An efficient, key-index-based, comparision based sorting algorithm.
+    It sorts based on the keys obtained on each index of the numbers to be sorteed.
+*/
+
+#include <iostream>
+#include <vector>
+#include <array>
+#include <algorithm>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+// The setFactors function sets the factors which depend on whether the sorting is needed to be done in ascending or descending order
+
+/*
+    count_sort
+    ----------
+    Performs Count Sort on the elements passed by the radix_sort function
+
+    Time complexity
+    ---------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+
+    Space complexity
+    ----------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    This space is used for count sorting each key and temporarily creating a sorted array.
+*/
+void count_sort(vector<int>& values, const int extractor, const bool to_show_state, const int mult_factor, const int add_factor) {
+    array<int, 10> counter{0};
+    vector<int> output(values.size());
+    for(int value : values)
+    	counter[add_factor + (mult_factor * ((value/extractor) % 10))]++;
```

Comment:
The above would be fine only if the expression was being used once, but it is used 3 times in this function, and so to reduce repetition (following the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)) we should use a function that returns the desired value. But to a function you would need to pass lots of variables to get the desired value. That is why I suggest using a lambda expression as the following:
```C++
int my_lambda = [&] (const int& value) {
    return add_factor + mult_factor * ((value / extractor) % 10);
};
```
Usage example:
```C++
for (const int& value : values)
    counter[my_lambda(value)]++;

...

for (int i = values.size() - 1; i >= 0; i--) {
    output[counter[my_lambda(values[i])] - 1] = values[i];
    ...
}
```
See, it's not at all complex! :smiley:

P. S. Remember to give the lambda a more meaningful name.